### PR TITLE
cmd/kubeapiserver: Clean code, avoid unnecessary condition, avoid non-nil but zero-length slice

### PR DESCRIPTION
**Pull request by tmilox**
Signed-off-by: TommyStarK <thomasmilox@gmail.com>

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

If applied this commit will clean a bit the code by avoiding unnecessary conditions,  avoid declaring non-nil but zero-length slice, use make to initialize map and name unused args with `_`

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by TommyStarK** [comment]
cc @liggitt, as usual any feedback from you will be greatly appreciated :)

File: cmd/kube-apiserver/app/aggregator.go [hunk 1]
```diff
@@ -112,7 +112,7 @@ func createAggregatorConfig(
 	}
 
 	// we need to clear the poststarthooks so we don't add them multiple times to all the servers (that fails)
-	aggregatorConfig.GenericConfig.PostStartHooks = map[string]genericapiserver.PostStartHookConfigEntry{}
```
Comment (liggitt) [inline cmd/kube-apiserver/app/aggregator.go:115]: I personally find the existing code style cleaner and easier to read

File: cmd/kube-apiserver/app/aggregator.go [hunk 2]
```diff
@@ -290,9 +290,10 @@ var apiVersionPriorities = map[schema.GroupVersion]priority{
 }
 
 func apiServicesToRegister(delegateAPIServer genericapiserver.DelegationTarget, registration autoregister.AutoAPIServiceRegistration) []*v1.APIService {
-	apiServices := []*v1.APIService{}
+	paths := delegateAPIServer.ListedPaths()
+	apiServices := make([]*v1.APIService, 0, len(paths))
 
-	for _, curr := range delegateAPIServer.ListedPaths() {
```
Comment (liggitt) [inline cmd/kube-apiserver/app/aggregator.go:295]: same here, I think the existing style is cleaner, and this is only executed once at process startup so the re-allocation of the slice is negligible

File: cmd/kube-apiserver/app/server.go [hunk 3]
```diff
@@ -654,16 +649,12 @@ func buildServiceResolver(enabledAggregatorRouting bool, hostname string, inform
 		return testServiceResolver
 	}
 
-	var serviceResolver webhook.ServiceResolver
+	serviceResolver := aggregatorapiserver.NewClusterIPServiceResolver(informer.Core().V1().Services().Lister())
```
Comment (liggitt) [inline cmd/kube-apiserver/app/server.go:652]: I think the existing code is more correct... rather than constructing and then overriding an implementation, we only construct the implementation we actually want

File: cmd/kube-apiserver/app/apiextensions.go [hunk 4]
```diff
@@ -39,7 +39,7 @@ import (
 func createAPIExtensionsConfig(
 	kubeAPIServerConfig genericapiserver.Config,
 	externalInformers kubeexternalinformers.SharedInformerFactory,
-	pluginInitializers []admission.PluginInitializer,
+	_ []admission.PluginInitializer,
```
Comment (liggitt) [inline cmd/kube-apiserver/app/apiextensions.go:42]: it's not very common to replace unused params with `_` in this repo... unless there's a specific reason we want to highlight this is unused, I would recommend reverting those changes

File: cmd/kube-apiserver/app/server.go [hunk 5]
```diff
@@ -673,19 +664,21 @@ func buildServiceResolver(enabledAggregatorRouting bool, hostname string, inform
...
+		apiServerServiceIP      net.IP
+		primaryServiceIPRange   net.IPNet
+		secondaryServiceIPRange net.IPNet
+	)
+
 	// nothing provided by user, use default range (only applies to the Primary)
 	if len(serviceClusterIPRangeList) == 0 {
 		var primaryServiceClusterCIDR net.IPNet
-		primaryServiceIPRange, apiServerServiceIP, err = controlplane.ServiceIPRange(primaryServiceClusterCIDR)
+		primaryServiceIPRange, apiServerServiceIP, err := controlplane.ServiceIPRange(primaryServiceClusterCIDR)
```
Comment (liggitt) [inline cmd/kube-apiserver/app/server.go:681]: if we're going to intentionally shadow the variables from the parent scope, I'd suggest moving the variable declarations below the `if len(serviceClusterIPRangeList) == 0 {` block to make it clear those are not related to this assignment

Comment (TommyStarK) [reply cmd/kube-apiserver/app/server.go:685]: `ServiceIPRange` expects a `net.IPNet` whereas `ParseCIDRSloppy` returns a `*net.IPNet` this is why I kept it as it was. I can revert back the whole if you prefer

**Comment by TommyStarK** [comment]
@liggitt I realize I'm starting to fall into the pattern of chasing contributions rather than offering less but with real improvements. I will take the time to find a topic where I can propose a relevant and useful change and avoid taking up your time unnecessarily. Thank you for everything.
ps: I can even go ahead and close this PR if you prefer.

Best regards,

**Comment by liggitt** [comment]
thanks for trying to balance churn of review/CI with net benefit of changes

in this PR, the `return kubeAPIServerConfig.Complete().New(delegateAPIServer)` cleanup and the os.Hostname reorder seem like clear improvements to readability... I'd probably drop the rest of the changes

**Comment by TommyStarK** [comment]
understood and done @liggitt
